integrations/http/http.go

```diff
@@ -171,15 +201,17 @@ func setBody(req *http.Request, rawBody string, formBody map[string]string, json
 		// This is required when using ioutil.NopCloser method for the request body
 		// (see ShouldSendChunkedRequestBody() in the library mentioned above).
 		req.ContentLength = int64(len(rawBody))
-		return nil
-	}
 
-	// JSON body.
-	if jsonBody.IsValid() && !jsonBody.IsNothing() {
-		if formBody != nil {
-			return errMutuallyExclusive
+		// Set the Content-Type header only if it's not already set.
+		if req.Header.Get(contentTypeHeader) == "" {
+			req.Header.Set(contentTypeHeader, "text/plain") // or "application/octet-stream"
 		}
+		return nil
 
+	case bodyTypeJSON:
+		if !jsonBody.IsValid() || jsonBody.IsNothing() {
+			return nil
```

Comment:
i think in this case the body content should be `null`

Reply:
could u please explain what you meant? 
I didn't change any logic here - it's copied from the original method - if json isn't valid (or empty) we are not setting the body.

Reply:
by default if not set, req.Body will be nil. you could see similar case in tests, which I added